boring/src/asn1.rs

```diff
@@ -504,22 +505,36 @@ impl Asn1ObjectRef {
     pub fn nid(&self) -> Nid {
         unsafe { Nid::from_raw(ffi::OBJ_obj2nid(self.as_ptr())) }
     }
-}
 
-impl fmt::Display for Asn1ObjectRef {
-    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+    /// Returns the numerical string OID of this object.
+    ///
+    /// This corresponds to [`OBJ_obj2txt`] with `no_name = 1`.
+    ///
+    /// [`OBJ_obj2txt`]: https://www.openssl.org/docs/man1.1.1/man3/OBJ_txt2obj.html
+    pub fn oid_string(&self) -> Result<String, Utf8Error> {
+        self.to_text(false)
+    }
+
+    // To promote this to `pub`, the call-site parameter meaning ought to be clearer
+    fn to_text(&self, no_name: bool) -> Result<String, Utf8Error> {
         unsafe {
             let mut buf = [0; 80];
             let len = ffi::OBJ_obj2txt(
                 buf.as_mut_ptr() as *mut _,
                 buf.len() as c_int,
                 self.as_ptr(),
-                0,
+                no_name as c_int,
             );
-            match str::from_utf8(&buf[..len as usize]) {
-                Err(_) => fmt.write_str("error"),
-                Ok(s) => fmt.write_str(s),
-            }
+            str::from_utf8(&buf[..len as usize]).map(|s| s.to_string())
+        }
+    }
+}
```

Comment:
What do you think about using `String::from_utf8_lossy` instead to avoid the Result return?

```rust
String::from_utf_lossy(&buf[..len as usize]).into_owned()
```

It's not like the caller can do anything if the string is invalid UTF-8, and this will at least provide *some* info. (A fully general operation might be something that transcodes with faux escape sequences, but that's probably overkill here.)

Reply:
I think this is reasonable! pushed it, along with the no_name fix.